nullaway/src/main/java/com/uber/nullaway/handlers/AbstractFieldContractHandler.java

```diff
@@ -0,0 +1,247 @@
+
+  /**
+   * Validates whether the parameter inside annotation conforms to the syntax rules. Parameters must
+   * conform to the following rules: 1. Cannot annotate a method with empty param set. 2. The
+   * receiver of selected fields in annotation can only be the receiver of the method. 3. All
+   * parameters given in the annotation must be one of the fields of the class or its super classes.
+   *
+   * @return Returns true, if the annotation conforms to the syntax rules.
+   */
+  protected boolean validateAnnotationSyntax(
+      Set<String> content,
+      NullAway analysis,
+      MethodTree tree,
+      VisitorState state,
+      Symbol.MethodSymbol methodSymbol) {
+    if (content == null) {
+      return true;
+    } else {
+      if (content.isEmpty()) {
+        // we should not allow useless annotations.
+        reportMatch(
+            analysis,
+            state,
+            tree,
+            "empty @"
+                + annotName
+                + " is the default precondition for every method, please remove it.");
+        return false;
+      } else {
+        for (String fieldName : content) {
+          if (fieldName.contains(".")) {
+            if (!fieldName.startsWith(THIS_NOTATION)) {
+              reportMatch(
+                  analysis,
+                  state,
+                  tree,
+                  "currently @"
+                      + annotName
+                      + " supports only class fields of the method receiver: "
+                      + fieldName
+                      + " is not supported");
+              return false;
+            } else {
+              fieldName = trimReceiver(fieldName);
+            }
+          }
+          Symbol.ClassSymbol classSymbol = ASTHelpers.enclosingClass(methodSymbol);
+          Element field = getFieldFromClass(classSymbol, fieldName);
+          if (field == null) {
+            reportMatch(
+                analysis,
+                state,
+                tree,
+                "cannot find field [" + fieldName + "] in class: " + classSymbol.getSimpleName());
+            return false;
+          }
+        }
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Returns the field name excluding its receiver (e.g. "this.a" will be "a")
+   *
+   * @param fieldName A class symbol.
+   * @return The class field name.
+   */
+  protected static String trimReceiver(String fieldName) {
+    return fieldName.substring(fieldName.lastIndexOf(".") + 1);
+  }
+
+  /**
+   * Finds a specific field of a class
+   *
+   * @param classSymbol A class symbol.
+   * @param name Name of the field.
+   * @return The class field with the given name.
+   */
+  protected static Element getFieldFromClass(Symbol.ClassSymbol classSymbol, String name) {
+    Preconditions.checkNotNull(classSymbol);
+    for (Element member : classSymbol.getEnclosedElements()) {
+      if (member.getKind().isField()) {
+        if (member.getSimpleName().toString().equals(name)) {
```

Comment:
Should we be checking here if the field is static?  I think we only want to match non-static fields

Reply:
How did you resolve the comment above?

Reply:
Out of curiosity: what was the resolution here? Was it really resolved? (I see it marked as resolved by @msridhar , so I assume we are fine with `static` fields after all?)

Reply:
As it turned out, we were not handling static fields correctly.  I just pushed https://github.com/uber/NullAway/pull/423/commits/62fb5cd97164ff52725a913ed943cb9050d6fe49 which at leasts reports an error when annotations mention a static field.  Are we ok with that @lazaroclapp, or do we want to add handling of static fields (which I think shouldn't be a huge problem)?  Or should we file a follow-up issue for that?

Reply:
Absolutely ok with actual handling happening as a follow-up. Let's get this landed.

That said, thanks for double-checking and adding the sanity check! :)